In [1]:
import pandas as pd
from sqlalchemy import create_engine
import nltk
import numpy as np
#nltk.download('stopwords')
#nltk.download('punkt')
#nltk.download('wordnet')

Import book_review and wiki_book_movie_ids_matching to filter out books

In [2]:
engine = create_engine('mysql+pymysql://dva:DVA2019!@dvaproject.c9f0lti9xqdg.us-east-1.rds.amazonaws.com/reviews?charset=utf8', echo=False) 

match = pd.read_sql("wiki_book_movie_ids_matching", con=engine)
book_rev = pd.read_sql("book_reviews", con=engine)
book_rev[0:5]
book_rev = pd.merge(book_rev, match[["id_goodreads", "BookTitle"]], left_on = "book_id", right_on = "id_goodreads") #match book_review with book-movie pair

In [3]:
book_rev

book_id   review_id           rating  \
0        915001   132633200  did not like it   
1        915001    95187570         liked it   
2        915001  1288028133   it was amazing   
3        915001    75229208  really liked it   
4        915001   142133155         liked it   
5        915001    25007208  really liked it   
6        915001   157669645  did not like it   
7        915001   895861893  really liked it   
8        915001    46782172   it was amazing   
9        915001  1918286637   it was amazing   
10       915001   996718431  really liked it   
11       915001  2038054481   it was amazing   
12       915001  1335883677   it was amazing   
13       915001    51029469   it was amazing   
14       915001   363586601   it was amazing   
15       915001    87824046  really liked it   
16       915001    95039293  really liked it   
17       915001   247290468   it was amazing   
18       915001    95002755   it was amazing   
19       915001   565548849         liked it   
20       915001   234615349         liked it   
21       915001  1925922612        it was ok   
22       915001  2255144003         liked it   
23       915001  2000833646   it was amazing   
24       915001  2055824387   it was amazing   
25       915001  1977192795   it was amazing   
26       915001  2003278040         liked it   
27       915001  2470085995  really liked it   
28       915001  2587636864  really liked it   
29       915001  2409834127   it was amazing   
...         ...         ...              ...   
382654   598330  1317343354         liked it   
382655   598330  1527786721         liked it   
382656   598330    84811701   it was amazing   
382657   598330   871026982   it was amazing   
382658   598330   110279422  really liked it   
382659   598330    49062594         liked it   
382660   598330   692073226  really liked it   
382661   598330  1761962651   it was amazing   
382662   598330  1747525317  really liked it   
382663   598330   407478057  really liked it   
382664   598330   142979017  really liked it   
382665   598330    71127456             None   
382666   598330  1333746717   it was amazing   
382667   598330  1486837174  really liked it   
382668   598330   192073253             None   
382669   598330   687254381         liked it   
382670   598330  1236135302         liked it   
382671   598330   593671763  really liked it   
382672   598330  1698372591         liked it   
382673   598330   102543942   it was amazing   
382674   598330  1381615051   it was amazing   
382675   598330   461834893  really liked it   
382676   598330   128595865  really liked it   
382677   598330   108304240  really liked it   
382678   598330   795073179         liked it   
382679   598330    74015125   it was amazing   
382680   598330    42349436  really liked it   
382681   598330   514022580  really liked it   
382682   598330  1406984395  really liked it   
382683   598330  1219559766   it was amazing   

                                              review_text language      id  \
0       Beaches was one of my all-time favorite movies...       en       1   
1       Having seen the movie Beaches more times than ...       en       2   
2       Excellent novel of friendship, filled with nos...       en       3   
3       loved it as much as i love the film. beautiful...       en       4   
4       **Contains adult language and content**(book I...       en       5   
5       I really enjoyed this book. The relationship b...       en       6   
6       Beaches is probably my all time favorite frien...       en       7   
7       I enjoyed reading this book. Beaches is one of...       en       8   
8       One of my all-time favorite chick flicks, I fo...       en       9   
9       This is one of my favorite books. Ever! For th...       en      10   
10      Long long time ago, in a faraway land, I was i...       en      11   
11      You are the wind beneath my wings..........Lei...       en      12   
12      When i ca

Basic cleaning

In [8]:
book_rev1= book_rev.loc[book_rev ['review_text'].notnull()].copy()   # when review text is not nan
#strip out all the punctuation
book_rev1['review_text'] = book_rev1.review_text.apply(lambda s: s.lower())
book_rev1['BookTitle'] = book_rev1.BookTitle.apply(lambda s: s.lower())
book_rev1['review_text'].replace({r'\'' : r''}, regex = True, inplace = True) #clean trailing dashes
book_rev1['BookTitle'].replace({r'\'' : r''}, regex = True, inplace = True) #clean trailing dashes
book_rev1['review_text'].replace({r'[^\x00-\x7f]' : r''}, regex = True, inplace = True) #remove all hex character
book_rev1['BookTitle'].replace({r'[^\x00-\x7f]' : r''}, regex = True, inplace = True) #remove all hex character
book_rev1['review_text'].replace({'â':'', '€':'', '¥':''}, regex = True, inplace = True) #further cleaning
book_rev1['BookTitle'].replace({'â':'', '€':'', '¥':''}, regex = True, inplace = True) #further cleaning
book_rev1['review_text'].replace({r'([^\s\w\-]|_)+' : r' '}, regex=True, inplace = True)
book_rev1['BookTitle'].replace({r'([^\s\w\-]|_)+' : r' '}, regex=True, inplace = True)

book_rev1['review_text'].replace({r'\--+|\\n|\s\s+' : r' '}, regex = True, inplace = True) #clean trailing dashes
book_rev1['BookTitle'].replace({r'\--+|\\n|\s\s+' : r' '}, regex = True, inplace = True) #clean trailing dashes
book_rev1 = book_rev1.loc[book_rev ['language']=='en']  #remove reivews written in another language
book_rev1['review_text'].replace({r'\s\-+\s|[\w\d]\-\s' : r''}, regex = True, inplace = True) #clean trailing dashes
book_rev1['BookTitle'].replace({r'\s\-+\s|[\w\d]\-\s' : r''}, regex = True, inplace = True) #clean trailing dashes
#book_rev1[0:10000].to_csv('book_rev_part1.csv')
#book_rev1[book_rev1['id_goodreads'] == 8127 ]
#book_rev1


In [9]:
#remove duplicate rows
book_rev1.drop_duplicates(subset=['book_id', 'review_id', "rating", "review_text", "language", "id_goodreads","BookTitle" ], keep="first", inplace=True)
book_rev1['BookTitle'].replace({r'\s\s+' : r' '}, regex = True, inplace = True) #clean all trailing spaces
book_rev1['BookTitle'].replace({r'\s\s+' : r' '}, regex = True, inplace = True) #clean all trailing spaces
#book_rev.drop([37233, 11082, 1250, 13065], inplace = True) #drop review without any words
#book_rev1[book_rev1['id_goodreads'] == 8127 ]
#book_rev[10000:20000].to_csv('book_rev_part2.csv')

Remove review's rating in review_text column

In [11]:
#stars = book_rev[book_rev.review_text.str.contains(r'\d\s\d\d.*\d+\sstars?')].copy()
#stars.to_csv('stars_rating.csv')
#book_rev[0:10000].to_csv('book_rev_part_before.csv')
#book_rev.drop(book_rev.index[[37233]], inplace = True)
#book_rev.index[book_rev['review_id']==2191718420]

In [16]:
import numpy as np
book_rev1['review_text'].replace({r'\d\s\d\d.*\d+\sstars?' : r''}, regex=True, inplace = True)
book_rev1['BookTitle'].replace({r'\d\s\d\d.*\d+\sstars?' : r''}, regex=True, inplace = True)
book_rev1['review_text'].replace({r'[0-9].*[0-9]\sstars?' : r''}, regex=True, inplace = True)  # clean format 3 0 to 3 5 stars
book_rev1['BookTitle'].replace({r'[0-9].*[0-9]\sstars?' : r''}, regex=True, inplace = True)  # clean format 3 0 to 3 5 stars
book_rev1['review_text'].replace({r'i would\s.*[\d]*stars?' : r''}, regex=True, inplace = True) # clean format i would__ stars
book_rev1['BookTitle'].replace({r'i would\s.*[\d]*stars?' : r''}, regex=True, inplace = True) # clean format i would__ stars
book_rev1['review_text'].replace({r'[A-Z0-9a-z-]*\s*stars?' : r''}, regex=True, inplace = True) # clean format 5 stars and five star
book_rev1['BookTitle'].replace({r'[A-Z0-9a-z-]*\s*stars?' : r''}, regex=True, inplace = True) # clean format 5 stars and five star
book_rev1['review_text'].replace({r'my?.*ratings?\:?.*\d' : r''}, regex=True, inplace = True) # clean rating
book_rev1['BookTitle'].replace({r'my?.*ratings?\:?.*\d' : r''}, regex=True, inplace = True) # clean rating
book_rev1['review_text'].replace({r'average rating of [\d.\s]* |[\w]*? rating:? |my\sratings?\:|rating:[\s\w*]*\d|\sof five|rating:' : r''}, regex=True, inplace = True) # clean 'my rating phrase'
book_rev1['BookTitle'].replace({r'average rating of [\d.\s]* |[\w]*? rating:? |my\sratings?\:|rating:[\s\w*]*\d|\sof five|rating:' : r''}, regex=True, inplace = True) # clean 'my rating phrase'
book_rev1['review_text'].replace({r'rating.*\d|\[rating.*\]|\d\s\d\/?\d' : r''}, regex=True, inplace = True) # clean 'my rating phrase'
book_rev1['BookTitle'].replace({r'rating.*\d|\[rating.*\]|\d\s\d\/?\d' : r''}, regex=True, inplace = True) # clean 'my rating phrase'
book_rev1['review_text'].replace({r'\(?paperbacks?\)?' : r''}, regex=True, inplace = True) # clean any discussion of paperback
book_rev1['BookTitle'].replace({r'\(?paperbacks?\)?' : r''}, regex=True, inplace = True) # clean any discussion of paperback
book_rev1['review_text'].replace({r'\s*?ratings|rating\s?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'\s*?ratings|rating\s?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'\shardcover\s?|hardcover|edition' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'\shardcover\s?|hardcover|edition' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'view spoiler|hide spoiler|spoilers?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'view spoiler|hide spoiler|spoilers?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'few\s|these\s|the\s|last\s|every\s|\d*\s?pages|\d*page\s*?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'few\s|these\s|the\s|last\s|every\s|\d*\s?pages|\d*page\s*?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'http\S+|www.\S+' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'http\S+|www.\S+' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'\s\s+' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'\s\s+' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'if.*link' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'if.*link' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'link to reviews?|click link|follow\slink\s|youtube\slinks?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'link to reviews?|click link|follow\slink\s|youtube\slinks?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'books?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'books?' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'films?' : r'movie'}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'films?' : r'movie'}, regex=True, inplace = True) # clean any mention of rating
book_rev1['review_text'].replace({r'(?!\b\d{4}\b|\b\d{2}\b)(?<!\d)\d+' : r''}, regex=True, inplace = True) # clean any mention of rating
book_rev1['BookTitle'].replace({r'(?!\b\d{4}\b|\b\d{2}\b)(?<!\d)\d+' : r''}, regex=True, inplace = True) # clean any mention of rating

#book_rev[book_rev.review_text == '']
book_rev1['review_text'].replace('', np.nan, inplace=True)
book_rev1.dropna(subset=['review_text'], inplace=True)
#book_rev1[0:10000].to_csv('book_rev_part.csv')
#book_rev[10000:20000].to_csv('book_rev_part2.csv')
#book_rev[book_rev['review_id'].str.contains('much')]
#book_rev1[1:1000]

In [17]:
book_rev1['review_text'] = book_rev1.apply(lambda x: x.review_text.replace( x.BookTitle, "" ), axis=1)


In [29]:
#book_rev1[1:1000]

Remove un-necessary story synposis from review_id = 142133155

In [19]:
book_rev1['review_text'] = book_rev1['review_text'].str.replace(r'contains\s.*synopsis\)','' )

In [20]:
tokens = book_rev1['review_text'].apply(nltk.word_tokenize)

Use this custom stopwords list

In [21]:
custom_words_1 = ["read","page","didnt", "lol","a","about","above","after","again","against","ain","all","am","an","and","any","are","aren","aren't","as","at","be","because","been","before","being","below","between","both","but","by","can","couldn","couldn't","d","did","didn","didn't","do","does","doesn","doesn't","doing","don","don't","down","during","each","few","for","from","further","had","hadn","hadnt","has","hasn","hasnt","have","haven","havent","having","he","her","here","hers","herself","him","himself","his","how","i","if","in","into","is","isn","isnt","it","its","its","itself","just","ll","m","ma","me","mightn","mightnt","more","most","mustn","mustnt","my","myself","needn","neednt","no","nor","not","now","o","of","off","on","once","only","or","other","our","ours","ourselves","out","over","own","re","s","same","shan","shant","she","shes","should","shouldve","shouldn","shouldnt","so","some","such","t","than","that","thatll","the","their","theirs","them","themselves","then","there","these","they","this","those","through","to","too","under","until","up","ve","very","was","wasn","wasnt","we","were","weren","werent","what","when","where","which","while","who","whom","why","will","with","won","wont","wouldn","wouldnt","y","you","youd","youll","youre","youve","your","yours","yourself","yourselves","could","hed","hell","hes","heres","hows","id","ill","im","ive","lets","ought","shed","shell","thats","theres","theyd","theyll","theyre","theyve","wed","well","were","weve","whats","whens","wheres","whos","whys","would","able","abst","accordance","according","accordingly","across","act","actually","added","adj","affected","affecting","affects","afterwards","ah","almost","alone","along","already","also","although","always","among","amongst","announce","another","anybody","anyhow","anymore","anyone","anything","anyway","anyways","anywhere","apparently","approximately","arent","arise","around","aside","ask","asking","auth","available","away","awfully","b","back","became","become","becomes","becoming","beforehand","begin","beginning","beginnings","begins","behind","believe","beside","besides","beyond","biol","brief","briefly","c","ca","came","cannot","cant","cause","causes","certain","certainly","co","com","come","comes","contain","containing","contains","couldnt","date","different","done","downwards","due","e","ed","edu","effect","eg","eight","eighty","either","else","elsewhere","end","ending","enough","especially","et","etc","even","ever","every","everybody","everyone","everything","everywhere","ex","except","f","far","ff","fifth","first","five","fix","followed","following","follows","former","formerly","forth","found","four","furthermore","g","gave","get","gets","getting","give","given","gives","giving","go","goes","gone","got","gotten","h","happens","hardly","hed","hence","hereafter","hereby","herein","heres","hereupon","hes","hi","hid","hither","home","howbeit","however","hundred","id","ie","im","immediate","immediately","importance","important","inc","indeed","index","information","instead","invention","inward","itd","itll","j","k","keep","keeps","kept","kg","km","know","known","knows","l","largely","last","lately","later","latter","latterly","least","less","lest","let","lets","like","liked","likely","line","little","ll","look","looking","looks","ltd","made","mainly","make","makes","many","may","maybe","mean","means","meantime","meanwhile","merely","mg","might","million","miss","ml","moreover","mostly","mr","mrs","much","mug","must","n","na","name","namely","nay","nd","near","nearly","necessarily","necessary","need","needs","neither","never","nevertheless","new","next","nine","ninety","nobody","non","none","nonetheless","noone","normally","nos","noted","nothing","nowhere","obtain","obtained","obviously","often","oh","ok","okay","old","omitted","one","ones","onto","ord","others","otherwise","outside","overall","owing","p","page","pages","part","particular","particularly","past","per","perhaps","placed","please","plus","poorly","possible","possibly","potentially","pp","predominantly","present","previously","primarily","probably","promptly","proud","provides","put","q","que","quickly","quite","qv","r","ran","rather","rd","readily","really","recent","recently","ref","refs","regarding","regardless","regards","related","relatively","research","respectively","resulted","resulting","results","right","run","said","saw","say","saying","says","sec","section","see","seeing","seem","seemed","seeming","seems","seen","self","selves","sent","seven","several","shall","shed","shes","show","showed","shown","showns","shows","significant","significantly","similar","similarly","since","six","slightly","somebody","somehow","someone","somethan","something","sometime","sometimes","somewhat","somewhere","soon","sorry","specifically","specified","specify","specifying","still","stop","strongly","sub","substantially","successfully","sufficiently","suggest","sup","sure","take","taken","taking","tell","tends","th","thank","thanks","thanx","thats","thatve","thence","thereafter","thereby","thered","therefore","therein","therell","thereof","therere","theres","thereto","thereupon","thereve","theyd","theyre","think","thou","though","thoughh","thousand","throug","throughout","thru","thus","til","tip","together","took","toward","towards","tried","tries","truly","try","trying","ts","twice","two","u","un","unfortunately","unless","unlike","unlikely","unto","upon","ups","us","use","used","useful","usefully","usefulness","uses","using","usually","v","value","various","ve","via","viz","vol","vols","vs","w","want","wants","wasnt","way","wed","welcome","went","werent","whatever","whatll","whats","whence","whenever","whereafter","whereas","whereby","wherein","wheres","whereupon","wherever","whether","whim","whither","whod","whoever","whole","wholl","whomever","whos","whose","widely","willing","wish","within","without","wont","words","world","wouldnt","www","x","yes","yet","youd","youre","z","zero","as","aint","allow","allows","apart","appear","appreciate","appropriate","associated","best","better","cmon","cs","cant","changes","clearly","concerning","consequently","consider","considering","corresponding","course","currently","definitely","described","despite","entirely","exactly","example","going","greetings","hello","help","hopefully","ignored","inasmuch","indicate","indicated","indicates","inner","insofar","itd","keep","keeps","novel","presumably","reasonably","second","secondly","sensible","serious","seriously","sure","ts","third","thorough","thoroughly","three","well","wonder"]
custom_words_2 = ["a","about","above","after","again","against","ain","all","am","an","and","any","are","aren","aren't","as","at","be","because","been","before","being","below","between","both","but","by","can","couldn","couldn't","d","did","didn","didn't","do","does","doesn","doesn't","doing","don","don't","down","during","each","few","for","from","further","had","hadn","hadn't","has","hasn","hasn't","have","haven","haven't","having","he","her","here","hers","herself","him","himself","his","how","i","if","in","into","is","isn","isn't","it","it's","its","itself","just","ll","m","ma","me","mightn","mightnt","more","most","mustn","mustnt","my","myself","needn","neednt","no","nor","not","now","o","of","off","on","once","only","or","other","our","ours","ourselves","out","over","own","re","s","same","shan","shant","she","shes","should","shouldve","shouldn","shouldnt","so","some","such","t","than","that","thatll","the","their","theirs","them","themselves","then","there","these","they","this","those","through","to","too","under","until","up","ve","very","was","wasn","wasnt","we","were","weren","werent","what","when","where","which","while","who","whom","why","will","with","won","wont","wouldn","wouldnt","y","you","youd","youll","youre","youve","your","yours","yourself","yourselves","could","hed","hell","hes","heres","hows","id","ill","im","ive","lets","ought","shed","shell","thats","theres","theyd","theyll","theyre","theyve","wed","well","were","weve","whats","whens","wheres","whos","whys","would","able","abst","accordance","according","accordingly","across","act","actually","added","adj","affected","affecting","affects","afterwards","ah","almost","alone","along","already","also","although","always","among","amongst","announce","another","anybody","anyhow","anymore","anyone","anything","anyway","anyways","anywhere","apparently","approximately","arent","arise","around","aside","ask","asking","auth","available","away","awfully","b","back","became","become","becomes","becoming","beforehand","begin","beginning","beginnings","begins","behind","believe","beside","besides","beyond","biol","brief","briefly","c","ca","came","cannot","cant","cause","causes","certain","certainly","co","com","come","comes","contain","containing","contains","couldnt","date","different","done","downwards","due","e","ed","edu","effect","eg","eight","eighty","either","else","elsewhere","end","ending","enough","especially","et","etc","even","ever","every","everybody","everyone","everything","everywhere","ex","except","f","far","ff","fifth","first","five","fix","followed","following","follows","former","formerly","forth","found","four","furthermore","g","gave","get","gets","getting","give","given","gives","giving","go","goes","gone","got","gotten","h","happens","hardly","hed","hence","hereafter","hereby","herein","heres","hereupon","hes","hi","hid","hither","home","howbeit","however","hundred","id","ie","im","immediate","immediately","importance","important","inc","indeed","index","information","instead","invention","inward","itd","itll","j","k","keep","keeps","kept","kg","km","know","known","knows","l","largely","last","lately","later","latter","latterly","least","less","lest","let","lets","like","liked","likely","line","little","ll","look","looking","looks","ltd","made","mainly","make","makes","many","may","maybe","mean","means","meantime","meanwhile","merely","mg","might","million","miss","ml","moreover","mostly","mr","mrs","much","mug","must","n","na","name","namely","nay","nd","near","nearly","necessarily","necessary","need","needs","neither","never","nevertheless","new","next","nine","ninety","nobody","non","none","nonetheless","noone","normally","nos","noted","nothing","nowhere","obtain","obtained","obviously","often","oh","ok","okay","old","omitted","one","ones","onto","ord","others","otherwise","outside","overall","owing","p","page","pages","part","particular","particularly","past","per","perhaps","placed","please","plus","poorly","possible","possibly","potentially","pp","predominantly","present","previously","primarily","probably","promptly","proud","provides","put","q","que","quickly","quite","qv","r","ran","rather","rd","readily","really","recent","recently","ref","refs","regarding","regardless","regards","related","relatively","research","respectively","resulted","resulting","results","right","run","said","saw","say","saying","says","sec","section","see","seeing","seem","seemed","seeming","seems","seen","self","selves","sent","seven","several","shall","shed","shes","show","showed","shown","showns","shows","significant","significantly","similar","similarly","since","six","slightly","somebody","somehow","someone","somethan","something","sometime","sometimes","somewhat","somewhere","soon","sorry","specifically","specified","specify","specifying","still","stop","strongly","sub","substantially","successfully","sufficiently","suggest","sup","sure","take","taken","taking","tell","tends","th","thank","thanks","thanx","thats","thatve","thence","thereafter","thereby","thered","therefore","therein","therell","thereof","therere","theres","thereto","thereupon","thereve","theyd","theyre","think","thou","though","thoughh","thousand","throug","throughout","thru","thus","til","tip","together","took","toward","towards","tried","tries","truly","try","trying","ts","twice","two","u","un","unfortunately","unless","unlike","unlikely","unto","upon","ups","us","use","used","useful","usefully","usefulness","uses","using","usually","v","value","various","ve","via","viz","vol","vols","vs","w","want","wants","wasnt","way","wed","welcome","went","werent","whatever","whatll","whats","whence","whenever","whereafter","whereas","whereby","wherein","wheres","whereupon","wherever","whether","whim","whither","whod","whoever","whole","wholl","whomever","whos","whose","widely","willing","wish","within","without","wont","words","world","wouldnt","www","x","yes","yet","youd","youre","z","zero","as","aint","allow","allows","apart","appear","appreciate","appropriate","associated","best","better","cmon","cs","cant","changes","clearly","concerning","consequently","consider","considering","corresponding","course","currently","definitely","described","despite","entirely","exactly","example","going","greetings","hello","help","hopefully","ignored","inasmuch","indicate","indicated","indicates","inner","insofar","itd","keep","keeps","novel","presumably","reasonably","second","secondly","sensible","serious","seriously","sure","ts","third","thorough","thoroughly","three","well","wonder","a", "about", "above", "above", "across", "after", "afterwards", "again", "against", "all", "almost", "alone", "along", "already", "also","although","always","am","among", "amongst", "amoungst", "amount", "an", "and", "another", "any","anyhow","anyone","anything","anyway", "anywhere", "are", "around", "as", "at", "back","be","became", "because","become","becomes", "becoming", "been", "before", "beforehand", "behind", "being", "below", "beside", "besides", "between", "beyond", "bill", "both", "bottom","but", "by", "call", "can", "cannot", "cant", "co", "con", "could", "couldnt", "de", "describe", "detail", "do", "done", "down", "due", "during", "each", "eg", "eight", "either", "eleven","else", "elsewhere", "empty", "enough", "etc", "even", "ever", "every", "everyone", "everything", "everywhere", "except", "few", "fifteen", "fify", "fill", "find", "fire", "first", "five", "for", "former", "formerly", "forty", "found", "four", "from", "front", "full", "further", "get", "give", "go", "had", "has", "hasnt", "have", "he", "hence", "her", "here", "hereafter", "hereby", "herein", "hereupon", "hers", "herself", "him", "himself", "his", "how", "however", "hundred", "ie", "if", "in", "inc", "indeed", "interest", "into", "is", "it", "its", "itself", "keep", "last", "latter", "latterly", "least", "less", "ltd", "made", "many", "may", "me", "meanwhile", "might", "mill", "mine", "more", "moreover", "most", "mostly", "move", "much", "must", "my", "myself", "name", "namely", "neither", "never", "nevertheless", "next", "nine", "no", "nobody", "none", "noone", "nor", "not", "nothing", "now", "nowhere", "of", "off", "often", "on", "once", "one", "only", "onto", "or", "other", "others", "otherwise", "our", "ours", "ourselves", "out", "over", "own","part", "per", "perhaps", "please", "put", "rather", "re", "same", "see", "seem", "seemed", "seeming", "seems", "serious", "several", "she", "should", "show", "side", "since", "sincere", "six", "sixty", "so", "some", "somehow", "someone", "something", "sometime", "sometimes", "somewhere", "still", "such", "system", "take", "ten", "than", "that", "the", "their", "them", "themselves", "then", "thence", "there", "thereafter", "thereby", "therefore", "therein", "thereupon", "these", "they", "thickv", "thin", "third", "this", "those", "though", "three", "through", "throughout", "thru", "thus", "to", "together", "too", "top", "toward", "towards", "twelve", "twenty", "two", "un", "under", "until", "up", "upon", "us", "very", "via", "was", "we", "well", "were", "what", "whatever", "when", "whence", "whenever", "where", "whereafter", "whereas", "whereby", "wherein", "whereupon", "wherever", "whether", "which", "while", "whither", "who", "whoever", "whole", "whom", "whose", "why", "will", "with", "within", "without", "would", "yet", "you", "your", "yours", "yourself", "yourselves", "the","a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p"]
custom_words = list(set(custom_words_1+custom_words_2))

In [22]:
stop_words = set(nltk.corpus.stopwords.words('english'))  
new_stop_word_list = stop_words.union(custom_words) # join default stopdword with custom stopword list

In [23]:
tokens = tokens.apply(lambda words: [w for w in words if w not in new_stop_word_list and len(w) > 2])

In [24]:
from nltk.stem.wordnet import WordNetLemmatizer
lemmatizer = WordNetLemmatizer()

In [25]:
lemmas = tokens.apply(lambda words: [lemmatizer.lemmatize(w) for w in words])

In [26]:
book_rev1["cleaned"] = lemmas

I convert it to CSV and then use R to upload it back to the server

In [27]:
book_rev1["cleaned"] = book_rev1["cleaned"].apply(lambda l: ' '.join(l))

In [ ]:
pd.set_option('display.max_colwidth', -1)
#book_rev1.review_text[book_rev1['cleaned'].str.contains("20198600588327")]

In [38]:
book_rev1.drop(columns = ['language', 'id_goodreads','BookTitle', 'review_id' ], inplace = True)
#book_rev1.drop(columns = ['rating' ], inplace = True)

In [40]:
book_rev1.to_csv('book_review_cleaned1.csv')